src/module/actor/mixins/actor-damage.js

```diff
@@ -657,10 +660,41 @@ export const ActorDamageMixin = (superclass) => class extends superclass {
 
         const originalCT = Math.floor((this.system.attributes.hp.max - originalHullPoints) / this.system.attributes.criticalThreshold.value);
         const newCT = Math.floor((this.system.attributes.hp.max - newHullPoints) / this.system.attributes.criticalThreshold.value);
+        let timesToRoll = 0;
+
         if (newCT > originalCT) {
```

Comment:
Lines 665 - 698 can be replaced with the following if we want to have rolls respect the rollMode selected in the sidebar, and override public to become a GM Roll for any effects applied to neutral, hostile, or secret tokens.

```suggestion
        // Make chat messages for critical threshold information gm-only if the token disposition is not Friendly.
        // Otherwise respect the rollMode setting in the sidebar
        let rollMode = game.settings.get("core", "rollMode");
        if (this.token.disposition < 1) {
            if (rollMode === "public") {
                rollMode = 'gmroll';
            }
        }

        if (newCT > originalCT) {
            const crossedThresholds = newCT - originalCT;
            const warningMessage = game.i18n.format("SFRPG.StarshipSheet.Damage.CrossedCriticalThreshold", {name: this.name, crossedThresholds: crossedThresholds});
            timesToRoll += crossedThresholds;
            ui.notifications.warn(warningMessage);
            const chatData = {
                user: game.user.id,
                speaker: ChatMessage.getSpeaker({actor: this}),
                content: warningMessage,
                type: CONST.CHAT_MESSAGE_STYLES.OTHER
            };

            ChatMessage.applyRollMode(chatData, rollMode);
            ChatMessage.create(chatData);
        }

        if (damage.isCritical && newHullPoints !== originalHullPoints) {
            timesToRoll++;
            const warningMessage = game.i18n.format((newCT > originalCT) ?  "SFRPG.StarshipSheet.Damage.Nat20WithThreshold" : "SFRPG.StarshipSheet.Damage.Nat20", {name: this.name});
            ui.notifications.warn(warningMessage);
            const chatData = {
                user: game.user.id,
                speaker: ChatMessage.getSpeaker({actor: this}),
                content: warningMessage,
                type: CONST.CHAT_MESSAGE_STYLES.OTHER
            };

            ChatMessage.applyRollMode(chatData, rollMode);
            ChatMessage.create(chatData);
        }

        if (timesToRoll > 0) {
            if (game.settings.get("sfrpg", "autoRollCritEffect")) {
                const pack = await game.packs.get('sfrpg.tables');
                const index = pack.index ?? await pack.getIndex();
                const obj = index.getName("Starship Critical Damage Effects");
                const doc = await pack.getDocument(obj._id);
                doc.drawMany(timesToRoll, {rollMode: rollMode});
            }
        }
```